Core/Handlers/MessageHandler.php

```diff
@@ -454,6 +472,28 @@ private function addCommonErrorHeadersToEnvelope(ErrorExchangeEnvelope $envelope
         $envelope->setHeader(ErrorExchangeEnvelope::HEADER_ERROR_NUM_RETRY, $retries);
     }
 
+    /**
+     * This method adds headers to the Envelope that we put the Callback exchange into, this is so that the
+     * consumer of the has information to do deal with it.
+     *
+     * - add the last error message and the time the error happened
+     * - add information about the processor that was being used when the event occurred
+     *
+     * @param CallbackExchangeEnvelope $envelope
+     * @param ProcessingException   $exception
+     */
+    private function addCallbackHeadersToEnvelope(CallbackExchangeEnvelope $envelope, ProcessingException $exception, ProcessorInterface $processor)
+    {
+        $originalException = $exception->getOriginalException();
+        $errorDescription = $originalException ? $originalException->getMessage() : $exception->getMessage();
+
+        $envelope->setHeader(CallbackExchangeEnvelope::HEADER_CREATED_AT, round(microtime(true) * 1000));
+        $envelope->setHeader(CallbackExchangeEnvelope::HEADER_ERROR_MESSAGE, $errorDescription);
```

Comment:
It is missing to add in the envelope the status code of the originalException. In this case needs to be mandatory from the original exception that basically it will be from "UnrecoverableErrorException"

Reply:
added